src/core/pile.sol

```diff
@@ -128,11 +128,20 @@ contract Pile is DSNote, TitleOwned {
         beans.drip(loans_[loan].fee);
     }
 
-    // repay() a certain amount of token from the user to the Pile
-    function repay(uint loan, uint wad) public owner(loan) note {
-        // moves currency from usr to pile and reduces debt
-        require(loans_[loan].balance == 0,"before repay loan needs to be withdrawn");
+
+    // recovery used for defaulted loans_
+    function recovery(uint loan, uint wad) public auth {
+        doRepay(loan, wad);
+
+        uint fee = loans_[loan].fee;
+        uint loss = beans.debtOf(loan, fee);
+
+        beans.decLoanDebt(loan, loans_[loan].fee, loss);
```

Comment:
why use `loans_[loan].fee` here if you already declared the variable above? or maybe we don't need to declare the variable.